CMakeLists.txt

```diff
@@ -27,6 +27,76 @@ ELSE()
     SET(VER_BUILD 0)
 ENDIF()
 
+# Configuration of compilators
+## Compiler flags to add
+### Flags for GCC and Clang
+set(GNU_FLAGS_WARNING_YES "-Wall -Wextra -Wformat=2 -Wpedantic -Wshadow -Wmissing-declarations -Wfloat-equal")
```

Comment:
CMake enforces less warning options, because it slows compiler (more things to check) and may warn too much for some bad written code. I created a lot of variables to split the different causes of the flags and give a name to causes.